packages/taquito-wallet-connect-2/src/taquito-wallet-connect-2.ts

```diff
@@ -0,0 +1,575 @@
+    const missingEvents: string[] = [];
+    requiredEvents.forEach((method) => {
+      if (!receivedEvents.includes(method)) {
+        missingEvents.push(method);
+      }
+    });
+    if (missingEvents.length > 0) {
+      this.clearState();
+      throw new InvalidReceivedSessionNamespace(
+        'All events must be approved',
+        getSdkError('USER_REJECTED_EVENTS').code,
+        'incomplete',
+        missingEvents
+      );
+    }
+  }
+
+  private validateAccounts(requiredNetwork: string[], receivedAccounts: string[]) {
+    if (receivedAccounts.length === 0) {
+      this.clearState();
+      throw new InvalidReceivedSessionNamespace(
+        'Accounts must not be empty',
+        getSdkError('USER_REJECTED_CHAINS').code,
+        'incomplete'
+      );
+    }
+    const receivedChains: string[] = [];
+    const invalidChains: string[] = [];
+    const missingChains: string[] = [];
+    const invalidChainsNamespace: string[] = [];
+
+    receivedAccounts.forEach((chain) => {
+      const accountId = chain.split(':');
+      if (accountId.length !== 3) {
+        invalidChains.push(chain);
+      }
+      if (accountId[0] !== TEZOS_PLACEHOLDER) {
+        invalidChainsNamespace.push(chain);
+      }
+      const network = accountId[1];
+      if (!receivedChains.includes(network)) {
+        receivedChains.push(network);
+      }
+    });
+
+    if (invalidChains.length > 0) {
+      this.clearState();
+      throw new InvalidReceivedSessionNamespace(
+        'Accounts must be CAIP-10 compliant',
+        getSdkError('USER_REJECTED_CHAINS').code,
+        'invalid',
+        invalidChains
+      );
+    }
+
+    if (invalidChainsNamespace.length > 0) {
+      this.clearState();
+      throw new InvalidReceivedSessionNamespace(
+        'Accounts must be defined in matching namespace',
+        getSdkError('UNSUPPORTED_ACCOUNTS').code,
+        'invalid',
+        invalidChainsNamespace
+      );
+    }
+    requiredNetwork.forEach((network) => {
+      if (!receivedChains.includes(network)) {
+        missingChains.push(network);
+      }
+    });
+    if (missingChains.length > 0) {
+      this.clearState();
+      throw new InvalidReceivedSessionNamespace(
+        'All chains must have at least one account',
+        getSdkError('USER_REJECTED_CHAINS').code,
+        'incomplete',
+        missingChains
+      );
+    }
+  }
+
+  private getTezosNamespace(): {
+    accounts: string[];
+    methods: string[];
+    events: string[];
+  } {
+    if (Object.prototype.hasOwnProperty.call(this.getSession().namespaces, TEZOS_PLACEHOLDER)) {
+      return this.getSession().namespaces[TEZOS_PLACEHOLDER];
+    } else {
+      throw new Error('invalid session, tezos namespace not found');
+    }
+  }
+
+  private getTezosRequiredNamespace(): {
+    chains: string[];
+    methods: string[];
+    events: string[];
+  } {
+    if (
+      Object.prototype.hasOwnProperty.call(this.getSession().requiredNamespaces, TEZOS_PLACEHOLDER)
```

Comment:
is there a reason why we needed to do the prototype inheritance `.calll()` instead of referencing `this.session` directly?

Reply:
Made the change